### PR TITLE
Switch to webview api

### PR DESCRIPTION
**Pull request by abe33**
Ref #172 

- [x] Status panel
- [x] Install flow
- [x] Fixing Install flow styles (@dhung09 maybe there's a pass to do here to make it consistent with atom)

### CODE REVIEW

File: src/install-webview.js [hunk 1]
```diff
@@ -4,8 +4,7 @@ const os = require('os');
 const vscode = require('vscode');
 const {workspace} = vscode;
 const formidable = require('formidable');
```
Comment (dbratz1177) [inline src/install-webview.js]: unused?

Comment (abe33) [reply src/install-webview.js]: nice catch, indeed we no longer need that as it was used to parse form data sent by the page when we had to rely on a server started by the extension.

**Comment by abe33** [comment]
I've just made a pass on the status view, the install flow will come after

> Should Kite Search go to the web docs (where it goes currently) or the copilot? All the other links go to the copilot. If we decide to keep it pointing to the webdocs, perhaps we should label it differently (e.g. Kite Web Docs)

I do agree this is definitely weird. Whatever solution is fine with me though. And even if we continue to point to the webdocs, do we still need the `desktopLogin` thing? (this is also true for Atom btw)

> We should not show Permissions in Kite Local, if possible

I guess this is a change we'll be able to address with kite-api 2.2 and your `isLocal` function. 

> When I hover on the links in the Status Panel, I get popups with what, from a user perspective, appears to be gobbledegook. I'm guessing we should probably get rid of that

Oh yeah I didn't noticed that mostly because I was clicking on them rather that hovering. Apparently it happens because we don't provide any `title` on the link. The solution here is to define that attribute for all the links, I'll make the change and we can change the wording later if needed. 

> There was no apparent action taken when I clicked Invite Friends
That was indeed broken, I missed these links when updating to `postMessage`. Now, these links were pointing to `http://localhost:46624/redirect/invite` and I get an error on `https://kite.com/invite` after the redirection. Is that still a thing?

**Comment by abe33** [comment]
@dbratz1177 For the install flow, when you talk about no action being done, are you talking about visual feedback? That was some of the things I was considering as improvements. 
As for the install not doing anything after providing the credentials I'm not sure what could happen, any error should move the flow to the error step (unless it's explicitly handled by the step, maybe that's what is broken) and in case we successfully authenticate on kite.com we start the download+install on a new view (the same install wait view that Daniel added for atom). I'll look into this.

**Comment by dbratz1177** [comment]
>That was indeed broken, I missed these links when updating to postMessage. Now, these links were pointing to http://localhost:46624/redirect/invite and I get an error on https://kite.com/invite after the redirection. Is that still a thing?
It's working for me (both the https://kite.com/invite and the redirection)

>For the install flow, when you talk about no action being done, are you talking about visual feedback?
After a second look, this is probably the essential issue that I was having... if we're requiring users to take action outside of VSCode to complete the installation process, I think we have to provide them some sort of feedback as to what to do next, and what to expect from the rest of the process.

**Comment by dbratz1177** [comment]
@abe33 I'm not aware of any issue with the backend that's generating the slowness.
the solution of having an explanatory intermediary screen seems good to me too

**Comment by dbratz1177** [comment]
@abe33 another option to an intermediary screen would be to just add a spinner like the installation page

**Pull request closed without merging**